### ISSUE TITLE
Error in instrumented process when Fuzzing `IR` method in Unit tests

### ISSUE THREAD

**Issue by alisevych** [UnitTestBot/UTBotJava#2545]
**Description**

Error in instrumented process when Fuzzing `IR` method in Unit tests

**To Reproduce**

1. Install [UnitTestBot plugin built from main, with Fuzzing in Unit tests support](https://github.com/UnitTestBot/UTBotJava/actions/runs/5935487385) in IntelliJ IDEA
2. Open `sm***t` project, `IR` class
3. Set Fuzzing 100% mode _//not strictly necessary_
4. Generate `Unit tests` for the last method in class with `@SpringBootApplication` selected

**Expected behavior**

No errors in instrumented process are expected.
Class is expected to be correctly instantiated and methods executed.

**Actual behavior**

There is error report with 15K-20K occurrences of `Default concrete execution failed`

**Screenshots, logs**

~~~java
public final class IrightResourceTest {
    public void testUn***t_errors() {
        // Couldn't generate some tests. List of errors:
        // 
        // 20559 occurrences of:
        // Default concrete execution failed

    }
}
~~~

**Environment**

IntelliJ IDEA version - 2023.2 Ultimate
Project - Maven
JDK - 1.8

**Comment by IlyaMuravjov** [comment]
After setting `Hanging test timeout` to `100 ms`, enabling `Power saving mode` on my laptop (to worsen the performance) and generating tests a few times, I've managed to get some concrete execution failures.

Here are the logs: [utbot-engine-current.log](https://github.com/UnitTestBot/UTBotJava/files/12430981/utbot-engine-current.log)

It seems that they were caused by timeout happening during evaluation of static initialization block in `JavaStdLibCustomModelConstructorsKt`, which caused class loader to mark `JavaStdLibCustomModelConstructorsKt` as non-loadable, since it can't be initialized, which in turn made all later invocations to fail with `NoClassDefFoundError: Could not initialize class JavaStdLibCustomModelConstructorsKt`.

That raises the question about of safe termination of hanging threads. I propose to do class loading and thread termination under same monitor to avoid such problems, however further discussion is advised.

**Comment by alisevych** [comment]
@tyuldashev 
The issue is not reproducing for me in UTBotJava with plugin built from latest main.
Can you please verify it?

**Comment by tyuldashev** [comment]
Yep, not reproducible at the moment. Though if probable cause is identified then we maybe should not close case without further discussion.
